sky/utils/cluster_yaml_utils.py

```diff
@@ -18,4 +18,7 @@ def get_provider_name(config: dict) -> str:
                                 provider_module)
     assert provider_search is not None, config
     provider_name = provider_search.group(1).lower()
+    # Special handling for lambda_cloud as Lambda cloud is registered as lambda.
+    if provider_name.endswith('_cloud'):
+        provider_name = provider_name.replace('_cloud', '')
```

Comment:
[Feel free to ignore] If we're special casing for lambda here, should we just match the exact string? Any clouds added in the future with `_cloud` in their name might also get caught here otherwise.
 
```suggestion
    if provider_name == 'lambda_cloud':
        provider_name = 'lambda'
```

Reply:
Good point! Adopted. Thanks!